src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -706,10 +706,10 @@ private string GeneratePropMetadataInitFunc(TypeGenerationSpec typeGenerationSpe
 
                 sb.Append($@"
 
-private static {JsonPropertyInfoTypeRef}[] {propInitMethodName}({JsonSerializerContextTypeRef} context)
+private {JsonPropertyInfoTypeRef}[] {propInitMethodName}({JsonSerializerContextTypeRef}? context)
 {{
-    {contextTypeRef} {JsonContextVarName} = ({contextTypeRef})context;
-    {JsonSerializerOptionsTypeRef} options = context.Options;
+    {contextTypeRef} {JsonContextVarName} = ({contextTypeRef}?)context ?? this;
```

Comment:
Not necessarily sure I understand the ramifications of this change. Who calls this method and under what circumstances can the context argument be null?

Reply:
What about the `if (context == null ...` checks? How were they being exercised before and did we have test coverage for them?

Reply:
I think it was just preventive check (probably triggered by nullability warning). It should not be possible for that ever to be null before combining was possible